scripts/kf7281/main.py

```diff
@@ -0,0 +1,188 @@
+
+    with open(filename, 'w') as configfile:
+        config.write(configfile)
+
+    return True
+
+def _single_repo_tics(repo_name: str, filename: Path):
+    """Create TIOBE scan workflow for a single charm repository.
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo_name: str, name of the repository / project to be used in TIOBE
+        filename: Path, name of the Github Action file
+    """
+
+    template = Path("./scripts/kf7281/tics-single-repo.yaml.j2")
+
+    env = jinja2.Environment()
+    with open(template, "r") as fid:
+        template = env.from_string(fid.read())
+
+    with open(filename, "w") as fid:
+        fid.write(template.render(
+            project_name=repo_name,
+            tics_auth_token="${{ secrets.TICSAUTHTOKEN }}"
+        ))
+
+
+def _multi_repo_tics(repo_name: str, charms: list[LocalCharmRepo], filename: Path):
+    """Create TIOBE scan workflow for a multi charm repository.
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo_name: str, name of the repository / project to be used in TIOBE
+        charms: list[kfcicli.charms.LocalCharmRepo], list of charms to be included in tiobe scanning
+        filename: Path, name of the Github Action file
+    """
+    template = Path("./scripts/kf7281/tics-multi-repo.yaml.j2")
+
+    env = jinja2.Environment()
+    with open(template, "r") as fid:
+        template = env.from_string(fid.read())
+
+    input_dict = {
+        charm.metadata.name: str(charm.tf_module.parent)
+        for charm in charms
+    }
+
+    with open(filename, "w") as fid:
+        fid.write(template.render(
+            project_name=repo_name,
+            tics_auth_token="${{ secrets.TICSAUTHTOKEN }}",
+            charms=input_dict
+        ))
+
+
+def create_tics_file(repo: Client, charms: list[LocalCharmRepo]):
+    """Create TIOBE scan workflow for a repository (either single-charm or multi-charm).
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo: kfcicli.repository.Client, client instance representing the repository
+        charms: list[kfcicli.charms.LocalCharmRepo], list of charms in the repository
+    """
+
+    filename = repo.base_path / ".github" / "workflows" / "tiobe_scan.yml"
+    if len(charms)==1:
+        _single_repo_tics(repo.base_path.name, filename)
+    else:
+        _multi_repo_tics(repo.base_path.name, charms, filename)
+
+
+def main(repo: Client, charms: list[LocalCharmRepo], dry_run: bool):
+    """Canon-run main function.
+
+    Signature must comply with the requirements of kfcicli.main.KubeflowCI.canon_run
+    """
+
+    charms_with_unit = []
+
+    for charm in charms:
+        if not (filename := charm.metadata.file.parent / "tox.ini").exists():
+            continue
+
+        reformat_tox(filename)
```

Comment:
We could have a comment in here mentioning why this is needed 
> * reformatting tox.ini file to be machine-generated and handled

Reply:
Done!